b_pull_Landsat_SRST_poi/py/gee_functions.py

```diff
@@ -0,0 +1,674 @@
+          .addBands(image.select(["Blue", "Green", "Red", 
+                                  "Nir", "Swir1", "Swir2", "SurfaceTemp"],
+                                ["sd_Blue", "sd_Green", "sd_Red", 
+                                "sd_Nir", "sd_Swir1", "sd_Swir2", "sd_SurfaceTemp"]))
+          .addBands(image.select(["Blue", "Green", "Red", "Nir", 
+                                    "Swir1", "Swir2", 
+                                    "SurfaceTemp"],
+                                  ["mean_Blue", "mean_Green", "mean_Red", "mean_Nir", 
+                                  "mean_Swir1", "mean_Swir2", 
+                                  "mean_SurfaceTemp"]))
+          .addBands(image.select(["SurfaceTemp"]))
+          .updateMask(d.eq(1)) # only high confidence water
+          .updateMask(r.eq(1)) #1 == no saturated pixels
+          .updateMask(f.eq(0)) #no snow or clouds
+          .updateMask(s.eq(0)) # no SR processing artefacts
+          .updateMask(hs.eq(1)) # only illuminated pixels
+          # add these bands back in to create summary statistics without the influence of the DSWE masks:
+          .addBands(pCount) 
+          .addBands(dswe1)
+          .addBands(dswe3)
+          .addBands(clouds) 
+          .addBands(hs)
+          .addBands(h)
+          ) 
+  combinedReducer = (ee.Reducer.median().unweighted().forEachBand(pixOut.select(["med_Blue", "med_Green", "med_Red", 
+            "med_Nir", "med_Swir1", "med_Swir2", "med_SurfaceTemp", 
+            "med_temp_qa","med_atran", "med_drad", "med_emis",
+            "med_emsd", "med_trad", "med_urad"]))
+    .combine(ee.Reducer.min().unweighted().forEachBand(pixOut.select(["min_SurfaceTemp", "min_cloud_dist"])), sharedInputs = False)
+    .combine(ee.Reducer.stdDev().unweighted().forEachBand(pixOut.select(["sd_Blue", "sd_Green", "sd_Red", "sd_Nir", "sd_Swir1", "sd_Swir2", "sd_SurfaceTemp"])), sharedInputs = False)
+    .combine(ee.Reducer.mean().unweighted().forEachBand(pixOut.select(["mean_Blue", "mean_Green", "mean_Red", 
+              "mean_Nir", "mean_Swir1", "mean_Swir2", "mean_SurfaceTemp"])), sharedInputs = False)
+    .combine(ee.Reducer.kurtosis().unweighted().forEachBand(pixOut.select(["SurfaceTemp"])), outputPrefix = "kurt_", sharedInputs = False)
+    .combine(ee.Reducer.count().unweighted().forEachBand(pixOut.select(["dswe_gt0", "dswe1", "dswe3"])), outputPrefix = "pCount_", sharedInputs = False)
+    .combine(ee.Reducer.mean().unweighted().forEachBand(pixOut.select(["clouds", "hillShadow"])), outputPrefix = "prop_", sharedInputs = False)
+    .combine(ee.Reducer.mean().unweighted().forEachBand(pixOut.select(["hillShade"])), outputPrefix = "mean_", sharedInputs = False)
+    )
+  # apply combinedReducer to the image collection, mapping over each feature
+  lsout = (pixOut.reduceRegions(feat, combinedReducer, 30))
+  out = lsout.map(remove_geo)
+  return out
+
+
+def ref_pull_89_DSWE1(image):
```

Comment:
The two function pairs for dswe 1 and 3 might be combined to save space and to avoid discrepancies when altering one not the other.

Reply:
I thought about this - but honestly I was getting stuck with limited py/GEE API chops to implement! If you have the capacity, I'd welcome thoughts about how to implement!

Reply:
I will see what I can do:D